tests/models/test_torchscript.py

```diff
@@ -109,9 +129,83 @@ def test_torchscript_properties(modelclass):
-    """ Test that scripted LightningModules is correctly saved and can be loaded. """
+    """ Test that scripted LightningModule is correctly saved and can be loaded. """
     model = modelclass()
     output_file = str(tmpdir / "model.pt")
     script = model.to_torchscript(file_path=output_file)
     loaded_script = torch.jit.load(output_file)
     assert torch.allclose(next(script.parameters()), next(loaded_script.parameters()))
+
+
+def test_torchcript_invalid_method(tmpdir):
+    """Test that an error is thrown with invalid torchscript method"""
+    model = BoringModel()
+    model.train(True)
+
+    with pytest.raises(ValueError, match="only supports 'script' or 'trace'"):
+        model.to_torchscript(method='temp')
+
+
+def test_torchscript_with_no_input(tmpdir):
+    """Test that an error is thrown when there is no input tensor"""
+    model = BoringModel()
+    model.example_input_array = None
+
+    with pytest.raises(ValueError, match='requires either `example_inputs` or `model.example_input_array`'):
+        model.to_torchscript(method='trace')
+
+
+@pytest.mark.skipif(
+    LooseVersion(torch.__version__) < LooseVersion("1.4.0"),
+    reason="torch has bug parsing namedtuples on torch < 1.4",
+)
+def test_torchscript_with_sequence_input(tmpdir):
+    """Test that traced LightningModule is created when input is tuple of tensors"""
+    class CustomModel(BoringModel):
+        def forward(self, x, y=None):
+            return super().forward(x)
+
+    def test_torchscript_export(model, example_inputs):
+        script = model.to_torchscript(example_inputs=example_inputs, method='trace')
+        assert isinstance(script, torch.jit.ScriptModule)
+
+        model.eval()
+        with torch.no_grad():
+            model_output = model(*example_inputs)
+            script_output = script(*example_inputs)
+        assert torch.allclose(script_output, model_output)
+
+    model = CustomModel()
+
+    # tuple input
+    example_inputs = (torch.randn(1, 32), torch.randn(1, 32))
+    test_torchscript_export(model, example_inputs)
+
+    # list input
+    example_inputs = [torch.randn(1, 32), torch.randn(1, 32)]
+    test_torchscript_export(model, example_inputs)
+
+    # NamedTuple input
+    example_inputs = namedtuple('sample', ['x', 'y'])
+    example_inputs = example_inputs(x=torch.randn(1, 32), y=torch.randn(1, 32))
+    test_torchscript_export(model, example_inputs)
+
+    with pytest.raises(ValueError, match='neither a Tensor nor tuple of Tensors'):
+        example_inputs = (torch.randn(1, 32), np.random.randn(1, 32))
+        test_torchscript_export(model, example_inputs)
+
+
+def test_error_with_invalid_input(tmpdir):
+    """Test that an error is thrown with invalid input"""
+    class CustomModel(BoringModel):
+        def forward(self, x):
+            if isinstance(x, dict):
+                x = x['x']
+
+            return super().forward(x)
+
+    model = CustomModel()
+    example_inputs = {'x': torch.randn(1, 32)}
```

Comment:
Why dict are not supported ?

Reply:
https://pytorch.org/docs/stable/generated/torch.jit.trace.html#torch.jit.trace
I think the reason might be that these script modules are meant to be platform-independent and should not necessary need a python dependency. And dict is only a python thing, but tuples are supported everywhere one or the other way.

Reply:
https://pytorch.org/docs/stable/jit_language_reference.html

It seems Dict are supported. 


<img width="748" alt="Screenshot 2020-11-18 at 08 30 20" src="https://user-images.githubusercontent.com/12861981/99505168-627b6380-2978-11eb-9293-74460c71ac12.png">